### PR TITLE
Introducing `stop` endpoint to stop running etcd-wrapper 

### PR DESCRIPTION
**Pull request by ishan16696**
**How to categorize this PR?**
/kind enhancement

**What this PR does / why we need it**:
- In order to achieve a restart of the etcd container inside the etcd member pod, this PR introduced a `/stop` endpoint in `etcd-wrapper`. Backup-restore can invoke this endpoint which will cause the container to exit and k8s to restart the container.

**Which issue(s) this PR fixes**:
Fixes https://github.com/gardener/etcd-wrapper/issues/24

**Special notes for your reviewer**:

**Release note**:

```other operator
Added a capability to stop the etcd-wrapper container by exposing an endpoint `/stop`. 
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,10 +2,25 @@
...
+IMAGE_REPOSITORY    := $(REGISTRY)/gardener/etcd-wrapper
+IMAGE_TAG           := $(VERSION)
+IMG ?= ${IMAGE_REPOSITORY}:${IMAGE_TAG}
+
 BIN_DIR := bin
 
 include hack/tools.mk
 
+.PHONY: docker-build
+docker-build:
```
Comment (unmarshall) [inline Makefile:16]: can you for now remove these targets as they are very g/g specific. Later we will add scripts to build and push docker images which will take in any registry as argument.

**Review by unmarshall — CHANGES_REQUESTED** [review]
just a minor comment, rest looks good.

File: Makefile (hunk 1 above)
Comment (ishan16696) [reply Makefile]: done. PTAL

**Review by unmarshall — APPROVED** [review]
/lgtm

**Comment by unmarshall** [comment]
I have tested the changes several times now.

**Pull request merged**